### DISCUSSION TITLE
Any way to add a custom "UniqueFor"?

### DISCUSSION THREAD

**Discussion by jamescrosswell** [VerifyTests/Verify#1029] [Q&A]
In the docs, it looks like [UniqueFor](https://github.com/VerifyTests/Verify/blob/main/docs/naming.md#uniquefor) can be used to add delimiters to the file names used... 

In our case, we want to run the same tests in two different scenarios:
- Default (Managed Code)
- AOT (Native Code)

We expect the resulting verified files to be different in those two different sceanrios, so I was looking for something in `VerifierSettings` that I could use to ensure the two sets of files would have different names.

So, we already have names like `MethodBeingTested.Run.DotNet7_0.verified.txt`... I'd like to have some additional tests run that generate some separate files called `MethodBeingTested.Run.DotNet7_0.Aot.verified.txt` or something like that.

Ideally this could be done in a `ModuleInitializer` once for our whole project/solution... rather than configuring it for every verify test we have individually. 

Is something like that possible? I didn't see any obvious solution looking at the docs or reading through discussions here.

**Comment by jamescrosswell** [comment]
Hm, I think I found a way to do this... do let me know if this is unadvisable though (I discovered this reading the source code - not the docs):

```csharp
    [ModuleInitializer]
    public static void Init()
    {
#if AOT_ENABLED
        Verifier.DerivePathInfo((sourceFile, projectDirectory, type, method) => new(
            directory: projectDirectory,
            typeName: type.Name,
            // This ensures a unique name for our verify files when performing AOT compilation
            methodName: method.Name + ".AOT"
            ));
#endif
    }
```

**Reply by SimonCropp** [reply]
yes derivepathinfo should work for you

and here is the docs https://github.com/VerifyTests/Verify/blob/main/docs/naming.md#derivepathinfo

**Comment by jamescrosswell** [comment]
Nice, thanks @SimonCropp !
